### PR TITLE
Use Pool2 instead of Generic-pool-redux

### PR DESCRIPTION
**Pull request by myndzi**
Passes sqlite and postgres integration tests, leaving it to Travis for the others

### CODE REVIEW

**Comment by myndzi** [comment]
This pull request is incompatible with the other one for binding error handlers for postgres:

https://github.com/tgriesser/knex/pull/659

Once this one is accepted, I will update that one.

**Comment by tgriesser** [comment]
Really awesome @myndzi! Let's kick the tires on this a bit and see if it causes any issues

@elliotf any chance you want to check this one out on master when you get a chance and see if it helps with issues you've also brought up previously regarding pooling

**Comment by myndzi** [comment]
He would also need https://github.com/tgriesser/knex/pull/659 just to be clear. Besides actively tracking all connections, pool2 provides fire-and-forget .remove and .destroy methods that work whether the resource is checked out or not. They still have to be called, though, which is what the other patch is for in the case of idle resources from pg.

File: lib/dialects/mysql/index.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ function connectionErrorHandler(client, connection, err) {
   if (connection && err && err.fatal) {
     if (connection.__knex__disposed) return;
     connection.__knex__disposed = true;
-    client.pool.genericPool.destroy(connection);
+    client.pool.destroy(connection);
```
Comment (elliotf) [inline lib/dialects/mysql/index.js]: should this be `destroy` or `release` like the others?

Comment (elliotf) [inline lib/dialects/mysql/index.js]: Ah, nevermind.  This happens on error; the others happen when they're done with the connection.

Comment (myndzi) [inline lib/dialects/mysql/index.js]: .release releases the resource to the pool
.remove removes the resource from the pool nicely
.destroy removes the resource from the pool dirtily

Knex only passes a remove handler so far so you should always call .remove

**Pull request merged**